packages/expo-updates/ios/EXUpdates/EXUpdatesAppController.m

```diff
@@ -251,36 +250,68 @@ - (void)appLoaderTask:(EXUpdatesAppLoaderTask *)appLoaderTask didFinishBackgroun
   }
 }
 
-# pragma mark - internal
+# pragma mark - EXUpdatesAppController+Internal
 
-- (EXUpdatesConfig *)_loadConfigFromExpoPlist
+- (BOOL)initializeUpdatesDirectoryWithError:(NSError ** _Nullable)error
 {
-  NSString *configPath = [[NSBundle mainBundle] pathForResource:EXUpdatesConfigPlistName ofType:@"plist"];
-  if (!configPath) {
-    @throw [NSException exceptionWithName:NSInternalInconsistencyException
-                                   reason:@"Cannot load configuration from Expo.plist. Please ensure you've followed the setup and installation instructions for expo-updates to create Expo.plist and add it to your Xcode project."
-                                 userInfo:@{}];
+  NSError *err;
+  _updatesDirectory = [EXUpdatesUtils initializeUpdatesDirectoryWithError:&err];
+  if (err && error) {
+    *error = err;
   }
+  return _updatesDirectory != nil;
+}
 
-  return [EXUpdatesConfig configWithDictionary:[NSDictionary dictionaryWithContentsOfFile:configPath]];
+- (BOOL)initializeUpdatesDatabaseWithError:(NSError ** _Nullable)error
+{
+  __block BOOL dbSuccess;
```

Comment:
Is there a scenario where `dbSucess = true`, but `dbError != null`?

Curios if we can get rid of this variable and just look at dbError?